src/tech/pipeline.md

```diff
@@ -0,0 +1,54 @@
+# The JIT pipeline.
+
+There are several trace representations used at different stages in Yorick's
+JIT pipeline.
+
+## PT Traces
+
+The JIT pipeline starts when a binary traces a section of its own execution,
+giving us a PT trace.
+
+A PT trace is a raw trace supplied to us by the Intel hardware (via the
+[hwtracer](https://github.com/softdevteam/hwtracer/) library). This kind of
+trace is effectively a list of block addresses. The addresses are virtual
+instruction addresses from the relocated `.text` section of the running binary.
+This means that (due to ASLR) these addresses are unique to a particular run of
+the program.
+
+## SIR Traces
+
+Given a PT trace, Yorick then constructs a SIR trace. A SIR trace is a list of
+<symbol-name, block-index> pairs. Each such pair identifies a SIR block that
+execution passed through during tracing.
+
+To make a SIR trace from a PT trace, Yorick uses the so-called "mapper". When
```

Comment:
Maybe:

...Yorick uses the [HWTMapper](https://github.com/softdevteam/yk/blob/master/yktrace/src/hwt/mapper.rs)